src/helpers/decoder.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable fp/no-mutation */
```

Comment:
Let's add a few docstrings and comments in this file to make it easier to come back to